### PR TITLE
Hide remote reindex migration behind feature flag

### PR DESCRIPTION
**Pull request by todvora**
## Description
Remote reindex migration will be enabled only when `remote_reindex_migration` is set to `on`

We will then add the following changes:

**For all users:**

- [x] „Methods for migration” block on welcome page is hidden when feature flag is set

**For users with incompatible versions:**

- [x] On the migration welcome page a comment is shown that they are using an incompatible version and that they need to upgrade to Opensearch to be able to use the migration wizard (only if feature flag is set)
- [x] Not able to proceed past welcome page if feature flag is set
- [x] This implies: No simple (i.e. without manual steps) way to migrate to an empty data node is available

**For users with compatible versions:**

- [x] No choice of migration method shown if feature flag is set, lead directly to in-place migration

## Motivation and Context
Implements https://github.com/Graylog2/graylog-plugin-enterprise/issues/8811

image

## How Has This Been Tested?
Manually, switching the feature flag

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by moesterheld — CHANGES_REQUESTED** [review]
Good solution, but there is still a part missing. This will leave users on Elasticsearch stuck in the wizard with no info why they can't proceed.
We either need to add a warning in the beginning that migration of data for these versions isn't supported and remove the guard or move the guard to the beginning of the wizard and also let them know in the UI.
And both the guard and the UI warning need to be aware of the feature flag

**Comment by todvora** [comment]
@moesterheld what about hiding the migration button/menu completely if you are on elasticsearch?

**Comment by moesterheld** [comment]
@todvora Hiding the tab completely would work, but would also take them the ability to move to data node without migrating any data. Some people might be happy to start with an empty Opensearch in data node.

**Review by todvora — COMMENTED** [review]
The backend part from @moesterheld looks good. I can't approve my own PR, but I tested it again locally.

File: graylog2-web-interface/src/components/datanode/migrations/MigrationWelcomeStep.tsx [hunk 1]
```diff
@@ -44,58 +47,72 @@ const StyledHelpPanel = styled(StyledPanel)`
...
+        </p>
+        <p>Deployments earlier than v5.2 or that opted to not install with a Data Node will need to migrate the message databases to Data Nodes.</p>
+        <p>
+          This migration tool will check the compatibility of your components and guide you through to migrate your existing OpenSearch data to a Data Node.<br />
+        </p>
+        <p>Migrating to Data Node will require some steps to be performed on the OS, within your current OS/ES cluster, and in your configuration files.</p>
+        <p>You can get more information on the Data Node migration <DocumentationLink page="graylog-data-node" text="documentation" />.</p>
+        <br />
+        <MigrationDatanodeList />
+        {!(isElasticsearch && !isRemoteReindexingEnabled) && (
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/migrations/MigrationWelcomeStep.tsx:75]:
```suggestion
        {(!isElasticsearch && !isRemoteReindexingEnabled) && (
```
makes it clearer I think

**Review by grotlue — APPROVED** [review]
Code-wise it looks good, one small comment and the Reviewbot also has some remarks. But don't want to block it because of that.

File: graylog2-web-interface/src/components/datanode/migrations/MigrationWelcomeStep.tsx (hunk 1 above)
Comment (gally47) [reply graylog2-web-interface/src/components/datanode/migrations/MigrationWelcomeStep.tsx:75]: that's not a typo it's the negation of (isElasticsearch && !isRemoteReindexingEnabled)

Comment (moesterheld) [reply graylog2-web-interface/src/components/datanode/migrations/MigrationWelcomeStep.tsx:75]: @grotlue it looks like that changes the logic

**Review by vvasylenko — APPROVED** [review]

**Comment by vvasylenko** [comment]
Looks good 👍

**Pull request merged**